Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -194,48 +218,38 @@ public final class RelayClient {
                 completion(nil)
             }
         }
-        cancellable = requestAcknowledgePublisher
-            .filter {$0.id == request.id}
-            .sink { (_) in
-                cancellable?.cancel()
-                completion(nil)
-            }
-        return request.id
-    }
-
-    private func setUpBindings() {
-        dispatcher.onMessage = { [weak self] payload in
-            self?.handlePayloadMessage(payload)
-        }
-        dispatcher.onConnect = { [unowned self] in
-            self.socketConnectionStatusPublisherSubject.send(.connected)
-        }
     }
 
+    // FIXME: Parse data to string once before trying to decode -> respond error on fail
     private func handlePayloadMessage(_ payload: String) {
-        if let request = tryDecode(SubscriptionRequest.self, from: payload), validate(request: request, method: .subscription) {
-            do {
-                try jsonRpcSubscriptionsHistory.set(topic: request.params.data.topic, request: request)
-                onMessage?(request.params.data.topic, request.params.data.message)
-                acknowledgeSubscription(requestId: request.id)
-            } catch {
-                logger.info("Relay Client Info: Json Rpc Duplicate Detected")
+        if let request = tryDecode(RPCRequest.self, from: payload) {
+            if let params = try? request.params?.get(Subscription.Params.self) {
+                do {
+                    try rpcHistory.set(request, forTopic: params.data.topic, emmitedBy: .remote)
```

Comment:
it is the only place where we are setting a history record. Should we record history when requesting as well?

Reply:
Right now, it's the only place. I'm in for recording all requests, so we can have a complete history.